### PR TITLE
Fixes user checks on pull() by implementing pull(user)

### LINKED ISSUE #8656: aconite unpullable by even non werewolves
### Labels

[BUG]

### Description

aconite flowers are unpullable by everyone, not just werewolves

### Steps to reproduce

makes aconite
try to pull it
cant pull it

### Expected Behavior

aconite should be pullable by non werewolves, and shortstun werewolves if they try to pull it

### Additional Information & Screenshots

Reported by: Balarak
On server: Goonstation 1 Classic: Heisenbee
Round log date: 2022-05-20-10-09
Reported on: 2022-05-20 11:26:30

### PR DESCRIPTION
**Pull request by MetricDuck**
[BUG]
## About the PR 
Several objects check various parameters about the user when someone attempts to pull said object. EG: Aconite checks to see if the user exists and that they are not werewolf otherwise it damages them.

However, the ```pull()``` parent doesn't define any arguments for ```pull(var/mob/user)```, so ```user``` will always be null. This effectively means that the checks will always fail: in some cases this meant that the object could never be pulled full stop as it would have an early return.

Accordingly this PR defines ```pull(src)``` within procs related to the relevant verbs (click and pull), allowing the user doing-the-pulling to be accessed as an argument elsewhere.

It touches every known usage of pull(), fixing the following checks:
- Aconite checking to see if the user exists or is a werewolf (and poisoning them);
- Slasher machete checking to see if the user should be allowed to drag the machete (or paining them);
- Staff of Cthulhu checking to see if it should allow the user to drag the staff (or frying them);
- Staff of Thunder checking to see if it should allow the user to drag the staff (or zapping them);
- The janitor holoemitter checking to see if it should allow the user to drag it (or warning them).

## Why's this needed? 
Fixes #8656

### CODE REVIEW

**Comment by Sovexe** [comment]
Really don't like `usr`, ever. Probably be a lot safer to just pass the user in the 6ish places `pull()` is called than to recode it all to trust `usr`

even the original define on `atom/movable` doesn't really need `usr`. It's just bad old code.

**Comment by MetricDuck** [comment]
> Really don't like `usr`, ever. Probably be a lot safer to...

Makes sense, following [http://www.byond.com/docs/ref/index.html#/proc/var/usr](url) I've now fixed it the other way by defining pull(src) in the relevant procs and then accessing src as an argument. I've tested each of the uses, there's not too many, and it all seems to work,

File: code/obj/item/mine.dm [hunk 1]
```diff
@@ -53,13 +53,13 @@
 
 		..()
 
-	pull()
+	pull(var/mob/user as mob)
```
Comment (Sovexe) [inline code/obj/item/mine.dm:56]: no need for `var/` or `as mob`. This goes for all of these
```suggestion
	pull(mob/user)
```

File: code/datums/hud/critter.dm [hunk 2]
```diff
@@ -263,7 +263,7 @@
 						to_pull = tgui_input_list(master, "Which do you want to pull? You can also Ctrl+Click on things to pull them.", "Which thing to pull?", pullable)
 					if(!isnull(to_pull) && BOUNDS_DIST(master, to_pull) == 0)
 						usr = master // gross
```
Comment (frawhst) [inline code/datums/hud/critter.dm:265]:
```suggestion
```
This line isn't necessary anymore
and is also awful. christ

File: code/datums/hud/human.dm [hunk 3]
```diff
@@ -477,7 +477,7 @@
 						to_pull = tgui_input_list(master, "Which do you want to pull? You can also Ctrl+Click on things to pull them.", "Which thing to pull?", pullable)
 					if(!isnull(to_pull) && BOUNDS_DIST(master, to_pull) == 0)
 						usr = master // gross
```
Comment (frawhst) [inline code/datums/hud/human.dm:479]:
```suggestion
```
.

File: code/datums/hud/robot.dm [hunk 4]
```diff
@@ -357,7 +357,7 @@
 						to_pull = tgui_input_list(master, "Which do you want to pull? You can also Ctrl+Click on things to pull them.", "Which thing to pull?", pullable)
 					if(!isnull(to_pull) && BOUNDS_DIST(master, to_pull) == 0)
 						usr = master // gross
```
Comment (frawhst) [inline code/datums/hud/robot.dm:359]:
```suggestion
```

File: code/datums/hud/critter.dm (hunk 2 above)
Comment (MetricDuck) [reply code/datums/hud/critter.dm]: Line removed

File: code/datums/hud/human.dm (hunk 3 above)
Comment (MetricDuck) [reply code/datums/hud/human.dm]: Line removed

File: code/datums/hud/robot.dm (hunk 4 above)
Comment (MetricDuck) [reply code/datums/hud/robot.dm]: Line removed, retested everything just in case and nothin broke!

**Pull request merged**